venues/auai.org/UAI/2019/Conference/python/register-pc-spc.py

```diff
@@ -0,0 +1,190 @@
+
+    registration_parent_invitation = client.post_invitation(openreview.Invitation(
+        id = conference.get_id() + '/-/Parent/Registration',
+        readers = ['everyone'],
+        writers = [conference.get_id()],
+        signatures = [conference.get_id()],
+        invitees = [conference.get_area_chairs_id(), conference.get_reviewers_id() ],
+        reply = {
+            'forum': None,
+            'replyto': None,
+            'readers': {'values': [conference.get_area_chairs_id(), conference.get_reviewers_id() ]},
+            'writers': {'values': [conference.get_id()]},
+            'signatures': {'values': [conference.get_id()]},
+            'content': {
+                'title': {'value': 'UAI 2019 Registration'},
+                'Subject Areas': {
+                    'value': subj_desc,
+                    'order': 1
+                },
+                'Profile Confirmed': {
+                    'value': coi_desc,
+                    'order': 2
+                },
+                'TPMS Account Confirmed': {
+                    'value': tpms_desc,
+                    'order': 3
+                }
+            }
+        }
+    ))
+
+    existing_reg_parents = client.get_notes(invitation=registration_parent_invitation.id)
+
+    if existing_reg_parents:
+        registration_parent_json = existing_reg_parents[0].to_json()
+    else:
+        registration_parent_json = {}
+
+    registration_parent_json.update({
+            'invitation': registration_parent_invitation.id,
+            'readers': registration_parent_invitation.reply['readers']['values'],
+            'writers': registration_parent_invitation.reply['writers']['values'],
+            'signatures': registration_parent_invitation.reply['signatures']['values'],
+            'replyto': None,
+            'forum': None,
+            'content': {
+                'title': registration_parent_invitation.reply['content']['title']['value'],
+                'Subject Areas': registration_parent_invitation.reply['content']['Subject Areas']['value'],
+                'Profile Confirmed': registration_parent_invitation.reply['content']['Profile Confirmed']['value'],
+                'TPMS Account Confirmed': registration_parent_invitation.reply['content']['TPMS Account Confirmed']['value'],
+            }
+        })
+
+    registration_parent = client.post_note(openreview.Note(**registration_parent_json))
+    print ('posted :', registration_parent.id)
+    # registration invitation
+    
+    registration_invitation = client.post_invitation(openreview.Invitation(
+        id = conference.get_id() + '/-/Registration',
+        duedate = tools.datetime_millis(datetime.datetime(2019, 3, 16, 10, 59)),
+        expdate = tools.datetime_millis(datetime.datetime(2019, 3, 16, 10, 59)),
```

Comment:
I wouldn't expire the invitation or I would expire it with a date ahead of the duedate.

Reply:
Shifted by a month.